src/ast/passes/semantic_analyser.cpp

```diff
@@ -949,13 +950,19 @@ void SemanticAnalyser::visit(Call &call)
                   },
           });
         }
-        std::string msg = validate_format_string(fmt.str, args);
+        std::string msg = validate_format_string(fmt.str, args, call.func);
         if (msg != "")
         {
           LOG(ERROR, call.loc, err_) << msg;
         }
       }
     }
+    if (call.func == "debugf" && is_final_pass())
+    {
+      LOG(WARNING, call.loc, out_)
+          << "The debugf() is highly disrecommended for production use, check "
+             "bpf_trace_printk in bpf-helpers(7).";
```

Comment:
```suggestion
          << "The debugf() builtin is not recommended for production use. For more information see "
             "bpf_trace_printk in bpf-helpers(7).";
```
Some re-wording that I think reads better (I don't think "disrecommend" is a word)